src/key_gen/mod.rs

```diff
@@ -60,32 +65,36 @@ impl From<Box<bincode::ErrorKind>> for Error {
 }
 
 /// A contribution by a node for the key generation. It must to be sent to all participating
-/// nodes and handled by all of them, including the one that produced it.
+/// nodes and handled by all of them, it contains `receiver_index, serialised row for the receiver,
+/// encrypted rows from the sender`.
 #[derive(Deserialize, Serialize, Clone, Hash, Eq, PartialEq, PartialOrd, Ord)]
-pub struct Part(BivarCommitment, Vec<Vec<u8>>);
+pub struct Part(u64, BivarCommitment, Vec<u8>, Vec<Vec<u8>>);
```

Comment:
It would be cleaner to change this to a struct with named fields.

Reply:
ok, will do